File: src/main/java/uk/gov/justice/hmpps/casenotes/repository/OffenderCaseNoteRepository.java
package uk.gov.justice.hmpps.casenotes.repository;

import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;
import uk.gov.justice.hmpps.casenotes.model.OffenderCaseNote;
import uk.gov.justice.hmpps.casenotes.model.SensitiveCaseNoteType;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Repository
public interface OffenderCaseNoteRepository extends PagingAndSortingRepository<OffenderCaseNote, UUID>, JpaSpecificationExecutor<OffenderCaseNote> {

    List<OffenderCaseNote> findAllByOffenderIdentifier(String offenderIdentifier);

    Long countOffenderCaseNotesByOffenderIdentifierAndSensitiveCaseNoteTypeAndOccurrenceDateTimeBetween(Long offenderIdentifier, SensitiveCaseNoteType type, LocalDate fromDate, LocalDate toDate);

}

File: src/main/java/uk/gov/justice/hmpps/casenotes/repository/OffenderCaseNoteRepository.kt
package uk.gov.justice.hmpps.casenotes.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import uk.gov.justice.hmpps.casenotes.domain.RefreshRepository
import uk.gov.justice.hmpps.casenotes.model.OffenderCaseNote
import java.util.UUID

@Repository
interface OffenderCaseNoteRepository :
  JpaSpecificationExecutor<OffenderCaseNote>,
  JpaRepository<OffenderCaseNote, UUID>,
  RefreshRepository<OffenderCaseNote, UUID>
